using-webhooks/client/android-java/app/src/main/java/com/example/app/CheckoutActivity.java

```diff
@@ -0,0 +1,170 @@
+                + "]"
+                + "}";
+        RequestBody body = RequestBody.create(json, mediaType);
+        Request request = new Request.Builder()
+                .url(backendUrl + "create-payment-intent")
+                .post(body)
+                .build();
+        httpClient.newCall(request)
+                .enqueue(new Callback() {
+                    @Override
+                    public void onFailure(Call call, IOException e) {
+                        runOnUiThread(() -> {
+                            Context applicationContext = getApplicationContext();
+                            Toast.makeText(applicationContext, "Error: " + e.toString(), Toast.LENGTH_LONG).show();
+                        });
+                    }
+
+                    @Override
+                    public void onResponse(Call call, final Response response) throws IOException {
+                        if (!response.isSuccessful()) {
+                            runOnUiThread(() -> {
+                                Context applicationContext = getApplicationContext();
+                                Toast.makeText(applicationContext, "Error: " + response.toString(), Toast.LENGTH_LONG).show();
+                            });
+                        } else {
+                            Gson gson = new Gson();
+                            Type type = new TypeToken<Map<String, String>>(){}.getType();
+                            Map<String, String> responseMap = gson.fromJson(response.body().string(), type);
+
+                            // The response from the server includes the Stripe public key and
+                            // PaymentIntent details.
+                            String stripePublishableKey = responseMap.get("publishableKey");
+                            paymentIntentClientSecret = responseMap.get("clientSecret");
+
+                            // Use the key from the server to initialize the Stripe instance.
+                            stripe = new Stripe(getApplicationContext(), stripePublishableKey);
+                        }
+                    }
+                });
+
+        // Hook up the pay button to the card widget and stripe instance
+        Button payButton = findViewById(R.id.payButton);
+        payButton.setOnClickListener((View view) -> {
+            CardInputWidget cardInputWidget = findViewById(R.id.cardInputWidget);
+            PaymentMethodCreateParams params = cardInputWidget.getPaymentMethodCreateParams();
+            if (params != null) {
+                ConfirmPaymentIntentParams confirmParams = ConfirmPaymentIntentParams
+                        .createWithPaymentMethodCreateParams(params, paymentIntentClientSecret);
+                stripe.confirmPayment(this, confirmParams);
+            }
+        });
+    }
+
+    private void displayAlert(Activity activity, String title, String message, boolean restartDemo) {
+        runOnUiThread(() -> {
```

Comment:
you shouldn't need to use `runOnUiThread` here, can you confirm that you do?

Reply:
thought I saw a crash or something, but I can't repro.  Removing.